server/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -388,20 +391,94 @@ public long valueBytes(Map<String, Object> map) {
         if (map == null) {
             return RamUsageEstimator.NUM_BYTES_OBJECT_HEADER;
         }
-        long bytes = RamUsageEstimator.shallowSizeOf(map);
-        long entrySize = -1;
-        for (var entry : map.entrySet()) {
-            if (entrySize == -1) {
-                entrySize = RamUsageEstimator.shallowSizeOf(entry);
+        return sizeOfMap(map);
```

Comment:
Okay for the fix, but longterm we should probably go with https://github.com/crate/crate/pull/16724#discussion_r1791570114